src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicListener.cs

```diff
@@ -240,8 +243,11 @@ private void Stop()
             {
                 ref NewConnectionInfo connectionInfo = ref *evt->Data.NewConnection.Info;
 
-                IPEndPoint localEndPoint = MsQuicAddressHelpers.INetToIPEndPoint(ref *(SOCKADDR_INET*)connectionInfo.LocalAddress);
-                IPEndPoint remoteEndPoint = MsQuicAddressHelpers.INetToIPEndPoint(ref *(SOCKADDR_INET*)connectionInfo.RemoteAddress);
+                Span<byte> localAddress = new Span<byte>((byte*)connectionInfo.LocalAddress, Internals.SocketAddress.IPv6AddressSize); // MsQuic always uses storage size as if IPv6
```

Comment:
I know this is just 2 lines per parameter, but having it hidden in a helper removes the conversion details from the new connection processing logic.